### PR TITLE
Changes to allow magic DNS provider to be specified (xip.io, nip.io, sslip.io)

### PR DESCRIPTION
**Pull request by bobdonat**
# Description

This pull request allows the magic DNS provider to be specified when installing Verrazzano.  

The verrazzano CRD has been changed to replace dns.XIPIO with dns.wildcard.domain where the DNS provider can be specified. For example:

    dns:
      wildcard:
        domain: nip.io

The install continues to default to xip.io for now.  That can be changed later if we want a different default.

Fixes VZ-2566

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Review by pfmackin — APPROVED** [review]

File: application-operator/controllers/ingresstrait/ingresstrait_controller.go [hunk 1]
```diff
@@ -733,19 +733,28 @@ func buildNamespacedDomainName(cli client.Reader, trait *vzapi.IngressTrait) (st
 
 	domain := externalDNSAnno[len(constants.VzConsoleIngress)+1:]
 
-	// If this is xip.io then build the domain name using Istio info
+	suffix := ""
 	if strings.HasSuffix(domain, "xip.io") {
-		domain, err = buildDomainNameForXIPIO(cli, trait)
+		suffix = "xip.io"
```
Comment (michael-cico) [inline application-operator/controllers/ingresstrait/ingresstrait_controller.go:738]: Don't have to do this on this pass, but maybe we should just validate this on the VZ CR in the webhook?  Then we can use the value without the if-branch.

Comment (bobdonat) [reply application-operator/controllers/ingresstrait/ingresstrait_controller.go:738]: yes, we can validate in the webhook but I think this check would still be needed.

File: application-operator/controllers/ingresstrait/ingresstrait_controller.go [hunk 2]
```diff
@@ -733,19 +733,28 @@ func buildNamespacedDomainName(cli client.Reader, trait *vzapi.IngressTrait) (st
...
+	} else if strings.HasSuffix(domain, "nip.io") {
```
Comment (gilbode) [inline application-operator/controllers/ingresstrait/ingresstrait_controller.go:739]: This should not be hardcoded to a predefined set of domains

Comment (bobdonat) [reply application-operator/controllers/ingresstrait/ingresstrait_controller.go:739]: working on fix in follow on PR

**Pull request merged**